ahrocksdb won't compile with librocksdb-dev 7 on Debian Bookworm

Issue: package won't compile on Debian Bookworm with librocksdb-dev provided in Debian's repo
How to repro: `opam install ahrocksdb` on Debian Bookworm

Error during compilation is below:

```
$ apt-cache policy librocksdb-dev
librocksdb-dev:
  Installed: 7.8.3-2
  Candidate: 7.8.3-2
  Version table:
 *** 7.8.3-2 500
        500 http://deb.debian.org/debian bookworm/main amd64 Packages
        100 /var/lib/dpkg/status
```

```
File "ffi/lib/dune", line 23, characters 0-155:
23 | (rule
24 |  (targets c_flags.sexp c_library_flags.sexp c_flags.txt c_library_flags.txt)
25 |  (deps    (:discover config/discover.exe))
26 |  (action  (run %{discover})))
(cd _build/default/ffi/lib && config/discover.exe)
failed to find an include path for RocksDB: are development headers installed on your system ?
tested paths: /usr/local/include/rocksdb /usr/include/rocksdb
compiling c program:
 | 
 | 
 | #include <c.h>
 | #include <version.h>
 | 
 | int main() {
 |   rocksdb_options_t* opt = rocksdb_options_create();
 |   rocksdb_options_destroy(opt);
 |   return 0;
 | };
 | 
run: gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC  -I/usr/local/include/rocksdb -I /home/szabolcs/ahrefs/monorepo/_opam/lib/ocaml -o /opam-tmp/build_d5ac95_dune/ocaml-configuratora3b6a9/c-test-0/test.exe /opam-tmp/build_d5ac95_dune/ocaml-configuratora3b6a9/c-test-0/test.c -lm -lrocksdb
-> process exited with code 1
-> stdout:
-> stderr:
 | /opam-tmp/build_d5ac95_dune/ocaml-configuratora3b6a9/c-test-0/test.c:3:10: fatal error: c.h: No such file or directory
 |     3 | #include <c.h>
 |       |          ^~~~~
 | compilation terminated.
compiling c program:
 | 
 | 
 | #include <c.h>
 | #include <version.h>
 | 
 | int main() {
 |   rocksdb_options_t* opt = rocksdb_options_create();
 |   rocksdb_options_destroy(opt);
 |   return 0;
 | };
 | 
run: gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC  -I/usr/include/rocksdb -I /home/szabolcs/ahrefs/monorepo/_opam/lib/ocaml -o /opam-tmp/build_d5ac95_dune/ocaml-configuratora3b6a9/c-test-2/test.exe /opam-tmp/build_d5ac95_dune/ocaml-configuratora3b6a9/c-test-2/test.c -lm -lrocksdb
-> process exited with code 1
-> stdout:
-> stderr:
 | In file included from /opam-tmp/build_d5ac95_dune/ocaml-configuratora3b6a9/c-test-2/test.c:4:
 | /usr/include/rocksdb/version.h:7:10: fatal error: string: No such file or directory
 |     7 | #include <string>
 |       |          ^~~~~~~~
 | compilation terminated.
Error: discover error
```

--- a/ffi/lib/config/discover.ml
+++ b/ffi/lib/config/discover.ml
@@ -26,7 +26,7 @@ int main() {
 |}
 in
 
-let c_flag = List.find_opt (fun c_flag -> C.c_test c ~c_flags:["-I" ^ c_flag] ~link_flags include_test) known_paths in
+let c_flag = List.find_opt (fun c_flag -> C.c_test c ~c_flags:["-I" ^ c_flag; "-x"; "c++"] ~link_flags include_test) known_paths in
 
 match c_flag with
 | None ->

